core/src/main/java/jenkins/model/Nameable.java

```diff
@@ -0,0 +1,19 @@
+package jenkins.model;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * An object that has a name.
+ * <p>
+ * This interface is used to provide a consistent way to retrieve the name of an object in Jenkins.
+ * It is typically implemented by objects that need to be identified by a name, such as tasks, nodes, or other model objects.
+ */
+public interface Nameable {
```

Comment:
Hmm. The spelling makes me grind my teeth. But https://www.merriam-webster.com/dictionary/nameable considers it “more common”. At least it matches https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Cloneable.html in terms of abusing the language.

Reply:
I initially chose `Named` but thought it may conflict with the annotation from `guice` and `javax.inject` (even though they are not used in Jenkins context, but they're still in classpath).

Reply:
FWIW I like `Named` better. These are objects which have a name, not objects which could be given a name.

Reply:
Ok, I can rename accordingly.